packages/core/src/utils.ts

```diff
@@ -72,6 +72,52 @@ export function throttle(
   }
 }
 
+const isContextArray = (value: ContextValue): value is ContextArray => Array.isArray(value)
+const isContext = (value: ContextValue): value is Context => !Array.isArray(value) && typeof value === 'object'
+
+/**
+ * Performs a deep merge of objects and arrays
+ * - arrays values are merged index by index
+ * - objects are merged by keys
+ * - values get replaced, unless undefined
+ *
+ * ⚠️ this method does not prevent infinite loops while merging circular references ⚠️
+ *
+ */
+export function deepMerge(destination: ContextValue, ...toMerge: ContextValue[]): ContextValue {
+  return toMerge.reduce((value1: ContextValue, value2: ContextValue): ContextValue => {
+    if (isContextArray(value1) && isContextArray(value2)) {
+      return [...Array(Math.max(value1.length, value2.length))].map((_, index) =>
+        deepMerge(value1[index], value2[index])
+      )
+    }
+    if (isContext(value1) && isContext(value2)) {
+      return Object.keys(value2).reduce(
```

Comment:
A for/in loop or a (polyfilled) usage of `Object.fromEntries` would be easier to understand and faster here.

Reply:
It's not really clear what's polyfilled and what is not 😅 And I'm not sure `ObjectEntries` would be faster as it'll be doing an extra `map` on the array

Reply:
It's not polyfilled. It would have a lower complexity, since you recreate the whole object for each key. Anyway, a for/in could do the trick :)